### DISCUSSION TITLE
vercel deployment fails with error  ERR_PNPM_EEXIST  EEXIST: file already exists

### DISCUSSION THREAD

**Discussion by seyaobey-dev** [vercel/community#6960] [Help]
### Summary

My vercel project is linked to my github project. The creation of a PR triggers a preview deployment. But now all deployments have started failing with below error. After the failure, if I manually restart the build (by ignoring the cache), the build and the deployment complete successfully. Below is the error thrown:

```bash
ERR_PNPM_EEXIST  EEXIST: file already exists, rename '/vercel/path0/node_modules/@parcel/watcher-wasm/node_modules' -> '/vercel/path0/node_modules/@parcel/watcher-wasm_tmp_142/node_modules'
```

This error occurs when vercel attempts to install dependencies with `pnpm install`. I have tried to disable the cache, to purge the cache. 

```bash
pnpm: 9.0.4,
node: 20.11.0
```

### Steps to Reproduce

The failing build is triggered by the creation of a new PR in github

**Comment by ryantbrown** [comment]
I have the same exact issue:

```
ERR_PNPM_EEXIST  EEXIST: file already exists, rename '/vercel/path0/node_modules/@parcel/watcher-wasm/node_modules' -> '/vercel/path0/node_modules/@parcel/watcher-wasm_tmp_135/node_modules'
--
21:35:56.177 | Error: Command "pnpm install" exited with 1
```

**Comment by ryantbrown** [comment]
The problem was the `pnpm` version, [Vercel supports 6, 7 and 8 ](https://vercel.com/docs/deployments/builds/package-managers) while I was using 9. Creating a `pnpm-lock.yaml` file using version 8 locally resolved the issue.

**Comment by ryangriffinau** [comment]
Ok, I also had the exact same error but node and pnpm packages were aligned. 
```
ERR_PNPM_EEXIST  EEXIST: file already exists, rename '/vercel/path0/node_modules/@parcel/watcher-wasm/node_modules' -> '/vercel/path0/node_modules/@parcel/watcher-wasm_tmp_142/node_modules'
```

What was not aligned were the dev and production env vars! Newly created env vars in dev weren't created in production and due to build cache in Vercel, it wasn't noticed until forcing a rebuild and receiving a server error.

My bad!

**Comment by kiikoh** [comment]
This error is still occuring

[a comment by an automated account is not shown]
